src/hooks/API.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const useApi = (url: string) => {
+
+  const [loading, setLoading] = useState(true);
+  const [data, setData] = useState([]);
+
+  const fetchApi = () => {
+    fetch(url)
+      .then((response) => {
+        return response.json();
+      })
+      .then((json) => {
+        setLoading(false);
+        setData(json);
+      });
```

Comment:
adds a catch? what if the response or json isn't there?

Reply:
before i set it to error but u said u didnt want to break the site ?

Reply:
You don't throw the error and break.
You return some status code or set some variable.
Have React monitor if a specific status is return, show Server 503 unavailable gracefully.

Reply:
i cant find any react component , is there any thing specific you have in mind ?